src/bootstrap/install.rs

```diff
@@ -251,6 +251,11 @@ install!((self, builder, _config),
     };
 );
 
+fn should_build_llvm(config: &Config) -> bool {
+    // additionaly check that llvm is not externally provided
+    LlvmTools::should_build(config) && config.is_rust_llvm(config.build)
```

Comment:
That code is not correct : the config.build is not the one we want...

Reply:
> if any of the configure_args items starts with --llvm-root

Can you do this check in here? It should do the work